client/src/components/LibraryNavigator/index.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA. All Rights Reserved.
+// Licensed under SAS Code Extension Terms, available at Code_Extension_Agreement.pdf
+
+import { commands, ExtensionContext, TreeView, window } from "vscode";
+import DataTable from "../../panels/DataTable";
+import DragAndDropController from "../DragAndDropController";
+import LibraryDataProvider from "./LibraryDataProvider";
+import LibraryModel from "./LibraryModel";
+import { LibraryItem, TableData } from "./types";
+
+class LibraryNavigator {
+  private libraryDataProvider: LibraryDataProvider;
+  private treeView: TreeView<LibraryItem>;
+
+  constructor(context: ExtensionContext) {
+    const dragAndDropController = new DragAndDropController<LibraryItem>(
+      context,
+      "application/vnd.code.tree.sas-library-navigator",
+      (item: LibraryItem | undefined) => (item.library ? item.uid : undefined)
+    );
+    this.libraryDataProvider = new LibraryDataProvider(
+      new LibraryModel(),
+      context.extensionUri
+    );
+    this.treeView = window.createTreeView("sas-library-navigator", {
+      treeDataProvider: this.libraryDataProvider,
+      dragAndDropController,
+    });
+
+    context.subscriptions.push(this.treeView);
+
+    commands.registerCommand(
```

Comment:
it is recommended to put registers into `context.subscriptions` to be properly disposed

Reply:
Good catch. I've updated `LibraryNavigator` and `ContentNavigator` to both implement a new `SubscriptionProvider` and tweaked how we setup these commands (see https://github.com/sassoftware/vscode-sas-extension/pull/143/commits/97877263840268c0408f02cdbf610a61691dc74b)